numbersofinterest/SeriesPropertiesTest.php

```diff
@@ -0,0 +1,69 @@
+<?php
+namespace NumbersOfInterest;
+require_once __DIR__ . '/SeriesProperties.php';
+use NumbersOfInterest\SeriesProperties;
+
+class SeriesPropertiesTest extends \PHPUnit_Framework_TestCase
+{
+    public function testWithAGoodInputIRetrieveAGoodSeriesProperties()
+    {
+        $series = new SeriesProperties('A;B;C', ';');
+
+        $this->assertEquals(
+            'totalOfNumbers:C, referenceNumber:B, occurencesOfReferenceNumber:A',
+            (string)$series
+        );
+    }
+
+    public function testWithAGoodInputIRetrieveGoodSeriesPropertiesAttributes()
+    {
+        $series = new SeriesProperties('A;B;C', ';');
+
+        $this->assertEquals(
+            'C',
+            $series->totalOfNumbers
+        );
+        $this->assertEquals(
+            'B',
+            $series->referenceNumber
+        );
+        $this->assertEquals(
+            'A',
+            $series->occurencesOfReferenceNumber
+        );
+    }
+
+    public function testWithFromAnInputArrayIRetrieveGoodSeriesPropertiesAttributes()
+    {
+        $series = SeriesProperties::fromArray(['A', 'B', 'C']);
+
+        $this->assertEquals(
+            'C',
+            $series->totalOfNumbers
+        );
+        $this->assertEquals(
+            'B',
+            $series->referenceNumber
+        );
+        $this->assertEquals(
+            'A',
+            $series->occurencesOfReferenceNumber
+        );
+    }
+    /**
+     * @expectedException InvalidArgumentException
+     */
+    public function testWhenSeriesPropertiesIsNotValidAnExceptionIsThrown()
+    {
+        new SeriesProperties('');
+    }
+
+    /**
+     * @expectedException InvalidArgumentException
+     */
+    public function testWhenSeriesPropertiesIsNotParsableWithSeparatorAnExceptionIsThrown()
```

Comment:
someone could say that the solution is overengineered due to the higher number of classes; I would try to organize them in folders separating:
- tests from production code
- domain from user interface (input and output management)